src/lib_ccx/ccx_encoders_spupng.c

```diff
@@ -1,411 +1,912 @@
-#include <assert.h>
-#include <sys/stat.h>
-#include "ccx_encoders_spupng.h"
+#include "ccfont2.xbm" // CC font from libzvbi
+#include "ccx_common_common.h"
+#include "ccx_encoders_common.h"
+#include "png.h"
+#include <ft2build.h>
+#include FT_FREETYPE_H
+#include "lib_ccx.h"
 #include "ccx_encoders_helpers.h"
+#include <assert.h>
+#ifdef ENABLE_OCR
+#include "ocr.h"
+#undef OCR_DEBUG
+#endif
+
+/* Closed Caption character cell dimensions */
+#define CCW 16
+#define CCH 26 /* line doubled */
+
+// TODO: To make them customizable
+#ifdef _WIN32
+#define FONT_PATH "C:\\Windows\\Fonts\\calibri.ttf"
+#else
+#define FONT_PATH "/usr/share/fonts/truetype/noto/NotoSans-Regular.ttf"
+#endif
```

Comment:
Missing font for macOS. Results in

Reply:
The font problem is actually one of the issue. I didn't fix it because I think it's part of customization. But yeah, I will fix it now, by adding an option.